src/ripple/consensus/ConsensusTypes.h

```diff
@@ -0,0 +1,234 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONSENSUS_CONSENSUS_TYPES_H_INCLUDED
+#define RIPPLE_CONSENSUS_CONSENSUS_TYPES_H_INCLUDED
+
+#include <ripple/basics/chrono.h>
+#include <ripple/consensus/ConsensusProposal.h>
+#include <ripple/consensus/DisputedTx.h>
+#include <chrono>
+#include <map>
+
+namespace ripple {
+
+/** Represents how a node currently participates in Consensus.
+
+    A node participates in consensus in varying modes, depending on how
+    the node was configured by its operator and how well it stays in sync
+    with the network during consensus.
+
+    @code
+      proposing               observing
+         \                       /
+          \---> wrongLedger <---/
+                     ^
+                     |
+                     |
+                     v
+                switchedLedger
+   @endcode
+
+   We enter the round proposing or observing. If we detect we are working
+   on the wrong prior ledger, we go to wrongLedger and attempt to acquire
+   the right one. Once we acquire the right one, we go to the switchedLedger
+   mode.  If we again detect the wrong ledger before this round ends, we go
```

Comment:
This reads weird: 

>Once we acquire the right one [...] If we again detect the wrong ledger [...] we go back [...] until we acquire the right one.

If we acquired the right one, how can we have the wrong one?